pkg/scm/bitbucket.go

```diff
@@ -0,0 +1,249 @@
+	if len(workspaces) > 1 {
+		prompt := &survey.Select{
+			Message: "Select the workspace:",
+			Options: workspaces,
+		}
+		if err := survey.AskOne(prompt, &workspace, survey.WithValidator(survey.Required)); err != nil {
+			return Context{}, err
+		}
+	}
+
+	pr, err := b.Client.Workspaces.Projects(workspace)
+	if err != nil {
+		return Context{}, err
+	}
+	projectKeys := make(map[string]string, 0)
+	projects := make([]string, 0)
+	for _, p := range pr.Items {
+		projectKeys[p.Name] = p.Key
+		projects = append(projects, p.Name)
+	}
+
+	project := ""
+	if len(projects) == 0 {
+		return Context{}, fmt.Errorf("You don't have any Bitbucket project created. Please create one first \n")
+	}
+
+	prompt := &survey.Select{
+		Message: "Select the project for your repo:",
+		Options: projects,
+	}
+	if err := survey.AskOne(prompt, &project, survey.WithValidator(survey.Required)); err != nil {
+		return Context{}, err
+	}
+
+	pub, priv, err := GenerateKeys(false)
+	if err != nil {
+		return Context{}, err
+	}
+
+	repoName, err := repoName()
+	if err != nil {
+		return Context{}, err
+	}
+
+	opt := &bitbucket.RepositoryOptions{
+		Uuid:      "",
+		Owner:     user.Username,
+		RepoSlug:  repoName,
+		Scm:       "git",
+		IsPrivate: "true",
+		Project:   projectKeys[project],
+	}
+
+	utils.Highlight("\ncreating bitbucket repository %s...\n", repoName)
+
+	res, err := b.Client.Repositories.Repository.Create(opt)
+	if err != nil {
+		return Context{}, err
+	}
+	sshAddress, err := getSSHAddress(res.Links)
+	if err != nil {
+		return Context{}, err
+	}
+
+	utils.Highlight("Setting up a read-write deploy key for this repo...\n")
+
+	if _, err := b.Client.Repositories.DeployKeys.Create(&bitbucket.DeployKeyOptions{
+		Owner:    user.Username,
+		RepoSlug: repoName,
+		Label:    "Plural Deploy Key",
+		Key:      pub,
+	}); err != nil {
+		return Context{}, err
+	}
+	// The bitbucket creates empty repository.
+	// Initialize bitbucket repo with .gitignore file.
+	dir, err := os.MkdirTemp("", "bitbucket")
+	if err != nil {
+		return Context{}, err
+	}
+	defer os.RemoveAll(dir)
+	gitIgnore := path.Join(dir, ".gitignore")
+	if err := os.WriteFile(gitIgnore, []byte(""), 0644); err != nil {
+		return Context{}, err
+	}
+
+	if err := b.Client.Repositories.Repository.WriteFileBlob(&bitbucket.RepositoryBlobWriteOptions{
+		Owner:    user.Username,
+		RepoSlug: repoName,
+		FilePath: gitIgnore,
+		FileName: ".gitignore",
+		Author:   fmt.Sprintf("%s <%s>", user.DisplayName, emailAddress),
+		Message:  "init",
+		Branch:   "master",
+	}); err != nil {
+		return Context{}, err
+	}
+
+	con := Context{}
+	con.pub = pub
+	con.priv = priv
+	con.username = user.Username
+	con.url = sshAddress
+	con.repoName = repoName
+	return con, nil
+}
+
+func (b *Bitbucket) StarPluralGitHubRep() error {
+	return nil
+}
+
+func getSSHAddress(links map[string]interface{}) (string, error) {
+	sshAddress := ""
+	var linkList []Link
+	if links != nil {
+		clone := links["clone"]
```

Comment:
what's going on here?  Seems very unusual we need to marshal then unmarshal whatever type this is

Reply:
god, that is bonkers